rasa_core_sdk/forms.py

```diff
@@ -251,8 +251,16 @@ def validate(self, dispatcher, tracker, domain):
         # extract requested slot
         slot_to_fill = tracker.get_slot(REQUESTED_SLOT)
         if slot_to_fill:
-            slot_values.update(self.extract_requested_slot(dispatcher,
-                                                           tracker, domain))
+            for slot, value in self.extract_requested_slot(dispatcher, tracker,
+                                                           domain).items():
+                validate_func = getattr(self, "validate_{}".format(slot),
+                                        lambda *x: value)
+                value = validate_func(value, dispatcher, tracker, domain)
+                if value:
+                    slot_values[slot] = value
+                else:
+                    tracker.slots[slot] = None
```

Comment:
it will not override slot value in `rasa_core`, in order to set the value to `None`, and action should return `SlotSet(slot, None)` event, so I think it makes sense to just write `slot_values[slot] = value` without `if`